truss/templates/control/control/helpers/truss_patch/requirement_name_identifier.py

```diff
@@ -1,10 +1,40 @@
+import logging
+import re
 from typing import Dict, List
+from urllib.parse import parse_qs, urlparse
 
 from packaging import requirements  # type: ignore
 
+logger: logging.Logger = logging.getLogger(__name__)
+
+URL_PATTERN = re.compile(r"^(https?|git|svn|hg|bzr)\+")
+
+
+def is_url_based_requirement(req: str) -> bool:
+    return bool(URL_PATTERN.match(req))
+
 
 def identify_requirement_name(req: str) -> str:
     try:
+        # parse as a url requirement
+        req = req.strip()
+        if is_url_based_requirement(req):
+            parsed_url = urlparse(req)
+            fragments = parse_qs(parsed_url.fragment)
+            if "egg" not in fragments:
+                logger.warning(
```

Comment:
what does this look like? Are there two log lines in this case? can we make sure there's only one

Reply:
got it. There's a bit of log noise to be fair. will update